addon-test-support/setup-global-a11y-hooks.ts

```diff
@@ -0,0 +1,38 @@
+import { _registerHook, HookUnregister } from '@ember/test-helpers';
+import { InvocationStrategy } from './types';
+import { getRunOptions } from './run-options';
+import { shouldForceAudit } from './should-force-audit';
+import a11yAudit from './audit';
+
+let _unregisterHooks: HookUnregister[] = [];
+
+/**
+ *
+ * @param shouldAudit Invocation strategy function that determines whether to run the audit helper or not.
+ * @param audit Optional audit function used to run the audit. Allows for providing either a11yAudit, a11yAuditIf,
+ *              or custom audit implementation.
+ */
+export function setupGlobalA11yHooks(
+  shouldAudit: InvocationStrategy,
+  audit: (...args: any[]) => PromiseLike<void> = a11yAudit
+) {
+  ['visit', 'click', 'doubleClick', 'tap'].forEach((helperName) => {
+    let hook = _registerHook(helperName, 'end', async () => {
+      if (shouldForceAudit() && shouldAudit(helperName, 'end')) {
```

Comment:
Curious as to the reason for the `shouldForceAudit()` check? As far as I can tell, this determines whether the `enableA11yAudit` query param is set. However, since the consumer can specify which variant to run (`a11yAudit`, `a11yAuditIf`, or custom method), shouldn't it be only relevant for the `a11yAuditIf` invocation? For `a11yAuditIf`, the check would effectively occur twice. For all other cases, we would need to either set the query param in the URL, or run `setEnableA11yAudit(true)` in the hooks.

Reply:
We want to deprecate `a11yAuditIf`. We can achieve that functionality in a different way, specifically though this new util.

Reply:
Yes, I get that. I just don't think `shouldForceAudit` should be a requisite for running `setupGlobalA11yHooks`, especially when using a custom audit implementation. Ultimately though, I'll leave this up to your discretion. Thanks!

Reply:
Without something like `shouldForceAudit` it is impossible for the addon to instruct folks how to reproduce a failure if they had used a `shouldAudit` that "randomizes". Having something that randomly fails tests, but is difficult/impossible to reproduce is **massively frustrating**. This `shouldForceAudit` does exactly that, it ensures that if you _want_ to run A11y audits, that you can reliably do it.

Reply:
Yeah, I guess I'm just thrown off with `?enableA11yAudit=true` being used for more than one purpose. Either way, you guys have way more insight into the testing infra space, so it's not something I'm gonna push.

Reply:
Ya, I definitely see what you mean there. I think that @scalvert and I are trying to move to a space where there is only one meaning (force running the audits). This includes a bunch of steps on the path though:

* stop suggesting actual test code use `a11yAudit` or `a11yAuditIf`
* remove the a11yAuditIf helper
* this implementation (for the hooks system)
* some additional implementation for customized validation of interaction helpers (fillIn, typeIn, &co) - likely a future PR
* move the non-testing code out of the addon (we totally need a nice dev mode experience, but it belongs in a separate addon)

Once all that stuff is done, I don't think we have that consistency problem where the query param means two things...